aiger_analysis/analysis.py

```diff
@@ -61,10 +51,26 @@ def fresh_var():
     return clauses, symbol_table, max_var
 
 
-def satisfiable(aig):
+def is_satisfiable(e):
     formula = CNF()
-    clauses, _, _ = tseitin(aig)
+    clauses, _, _ = _tseitin(e)
     for clause in clauses:
         formula.append(clause)
     with Lingeling(bootstrap_with=formula.clauses) as ling:
         return ling.solve()
+
+
+def is_valid(e):
+    aig = cmn.extract_aig(e)
+    aig = aig >> aiger.bit_flipper(inputs=aig.outputs)  # negate
```

Comment:
Actually needs to be
`e = ~BoolExpr(cmn.extract_aig(e))`
Because e can be aiger already. The type that is_satisfiable is called with does not matter.

Reply:
Makes sense.